--- a/libavcodec/mediacodecenc.c
+++ b/libavcodec/mediacodecenc.c
@@ -121,8 +121,8 @@ static const struct {
 
 static const enum AVPixelFormat avc_pix_fmts[] = {
     AV_PIX_FMT_MEDIACODEC,
-    AV_PIX_FMT_YUV420P,
     AV_PIX_FMT_NV12,
+    AV_PIX_FMT_YUV420P,
     AV_PIX_FMT_NONE
 };
 
@@ -517,8 +517,14 @@ static av_cold int mediacodec_init(AVCodecContext *avctx)
     if (ret != COLOR_TRANSFER_UNSPECIFIED)
         ff_AMediaFormat_setInt32(format, "color-transfer", ret);
 
-    if (avctx->bit_rate)
+    if (avctx->bit_rate) {
         ff_AMediaFormat_setInt32(format, "bitrate", avctx->bit_rate);
+    } else {
+        int64_t dbr = (int64_t)s->width * s->height * s->fps * 0.1;
+        if (dbr < 1000000) dbr = 2000000;
+        if (dbr > 8000000) dbr = 8000000;
+        ff_AMediaFormat_setInt32(format, "bitrate", (int32_t)dbr);
+    }
     if (s->bitrate_mode >= 0) {
         ff_AMediaFormat_setInt32(format, "bitrate-mode", s->bitrate_mode);
         if (s->bitrate_mode == BITRATE_MODE_CQ && avctx->global_quality > 0)
