--- /tmp/extract/libva-2.21.0/va/meson.build.orig	2026-07-13 16:23:22.248309501 -0500
+++ /tmp/extract/libva-2.21.0/va/meson.build	2026-07-13 16:23:22.272309128 -0500
@@ -58,12 +58,8 @@
 libva_sym = 'libva.syms'
 libva_sym_arg = '-Wl,-version-script,' + '@0@/@1@'.format(meson.current_source_dir(), libva_sym)
 
-libva_link_args = []
-libva_link_depends = []
-if cc.links('', name: '-Wl,--version-script', args: ['-shared', libva_sym_arg])
-  libva_link_args = libva_sym_arg
-  libva_link_depends = libva_sym
-endif
+libva_link_args = libva_sym_arg
+libva_link_depends = libva_sym
 
 install_headers(libva_headers, subdir : 'va')
 
