--- /tmp/extract/Vulkan-Loader-vulkan-sdk-1.3.280.0/loader/CMakeLists.txt.orig	2026-07-13 16:24:30.341250309 -0500
+++ /tmp/extract/Vulkan-Loader-vulkan-sdk-1.3.280.0/loader/CMakeLists.txt	2026-07-13 16:24:30.364249951 -0500
@@ -365,12 +365,17 @@
         mark_as_advanced(APPLE_STATIC_LOADER)
     endif()
 
-    if(APPLE_STATIC_LOADER)
+    option(BUILD_STATIC_LOADER "Build a static Vulkan loader (libvulkan.a) for non-Apple platforms." OFF)
+
+    if(APPLE_STATIC_LOADER OR BUILD_STATIC_LOADER)
         add_library(vulkan STATIC)
-        target_compile_definitions(vulkan PRIVATE APPLE_STATIC_LOADER)
 
-        message(WARNING "The APPLE_STATIC_LOADER option has been set. Note that this will only work on MacOS and is not supported "
-                "or tested as part of the loader. Use it at your own risk.")
+        if(APPLE_STATIC_LOADER)
+            target_compile_definitions(vulkan PRIVATE APPLE_STATIC_LOADER)
+
+            message(WARNING "The APPLE_STATIC_LOADER option has been set. Note that this will only work on MacOS and is not supported "
+                    "or tested as part of the loader. Use it at your own risk.")
+        endif()
     else()
         add_library(vulkan SHARED)
     endif()
